### DISCUSSION TITLE
Toast top action navigates to home page

### DISCUSSION THREAD

**Discussion by Brokemia** [microsoft/fluentui-blazor#2113] [Q&A] [answered]
I'm trying to create toasts with a top action that basically acts as an undo button for the thing that triggered the toasts. However, clicking the action seems to redirect me back to the home page or more specifically, back to https://localhost:7077/#

Here's a significantly slimmed down version of the code I have (that I used to test this issue), but you'll see there's nothing fancy here.
```razor
@page "/"
@inject IToastService service
@rendermode InteractiveServer

<PageTitle>Home</PageTitle>

<h1>Hello, world!</h1>

Welcome to your new app.
<FluentButton OnClick=test >Test</FluentButton>


@code {
    private void test() {
        Console.WriteLine("hi");
        service.ShowSuccess("test", 10_000_000, "Do thing", EventCallback.Factory.Create<ToastResult>(this, test));
    }
}
```

I also have `<FluentToastProvider @rendermode="InteractiveServer" />` in my layout file.

How can I avoid this redirect?

**Comment by vnbaaij** [comment]
The last parameter is the callback function you want to be called when the action is clicked. See https://fluentui-blazor.net/Toast#toast with  sync and async example for the general ShowToast method where a message is  logged

**Reply by Brokemia** [reply]
Yes, I've been using that as a reference. Using the same method for both the click and callback was just for testing. Even with two separate methods, like so:
```cs
private void test() {
    Console.WriteLine("hi");
    service.ShowToast(ToastIntent.Success, "test", 10_000_000, "Do thing", EventCallback.Factory.Create<ToastResult>(this, callback));
}

private void callback(ToastResult result) {
    Console.WriteLine("callback");
}
```

I get the same result. Nothing is printed from the callback, and it's like I clicked on a link to the homepage (but with a # after). I can't see any reason why my code is failing in this way, but the demo code works

**Reply by vnbaaij** [reply]
We can't really assess what is happening without any source to look at. Maybe create a repo we can look at?

**Reply by Brokemia** [reply]
Sure, I put my code at [https://github.com/Brokemia/.NET-CrowdDoc](https://github.com/Brokemia/.NET-CrowdDoc/tree/de53336b9781d7de101edb6090eb7fde2e75a9c1)

I put the test code in [https://github.com/Brokemia/.NET-CrowdDoc/blob/main/Components/Pages/ToastTest.razor](https://github.com/Brokemia/.NET-CrowdDoc/blob/de53336b9781d7de101edb6090eb7fde2e75a9c1/Components/Pages/ToastTest.razor)

If you run it, you'll see that clicking the action on the toast takes you to the home page, which is not what I want

**Reply by vnbaaij** [reply]
I'm getting errors when trying to run the cloned project. Something with auth/clientid.

**Reply by Brokemia** [reply]
Ah, sorry, you can put
```json
{
  "Authentication:Google:ClientId": "test",
  "Authentication:Google:ClientSecret": "test"
}
```
in the secrets.json file (under Connected Services), and it should load fine

(also I accidentally put this as an answer, in case you got double emails)

**Reply by vnbaaij** [reply, marked as answer]
Ok, I got the solution running.

The issue is caused by interactivity not being fully available in the app on the toast page, even though you've specified a rendermode on the page itself.  If you change line 31 in `App.razor` to `<Routes @rendermode=InteractiveServer />` It starts working like it should.

So my conclusion would be toast/SSR do not play nice together. Just adding the rendermode to the provider is not enough

**Reply by oneolddev** [reply]
I believe it's related to the following known issues:
- Ability to add rendermode to blazor layouts [#52709](https://github.com/dotnet/aspnetcore/issues/52709)
- [Blazor] Support serializing render fragments from SSR [#52768](https://github.com/dotnet/aspnetcore/issues/52768)

**Reply by Brokemia** [reply]
Thanks for the help with this

**Reply by Brokemia** [reply]
A small addition, just for completeness, turning on InteractiveServer mode for my Routes component broke my login page, so I had to follow [this Stack Overflow solution](https://stackoverflow.com/a/78126190/6337971) to fix it

**Reply by vnbaaij** [reply]
Just for the record, InteractiveServer breaking login page is not a library thing though. It is a general Blazor issue.
